e2e/config/config_test.go

```diff
@@ -254,3 +269,86 @@ func TestConfigNotEqual(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateDeployers(t *testing.T) {
+	tests := []struct {
+		name   string
+		config *Config
+		valid  bool
+	}{
+		{
+			name: "valid",
+			config: &Config{
+				Deployers: []Deployer{
+					{Name: "appset", Type: "appset", Description: "ApplicationSet deployer for ArgoCD"},
+					{Name: "subscr", Type: "subscr", Description: "Subscription deployer for OCM subscriptions"},
+					{Name: "disapp", Type: "disapp", Description: "Discovered Application deployer for discovered applications"},
+				},
+			},
+			valid: true,
+		},
+		{
+			name: "empty",
+			config: &Config{
+				Deployers: []Deployer{},
+			},
+			valid: false,
+		},
+		{
+			name: "duplicate names",
+			config: &Config{
+				Deployers: []Deployer{
+					{Name: "appset", Type: "appset", Description: "ApplicationSet deployer for ArgoCD"},
+					{Name: "appset", Type: "appset", Description: "Duplicate ApplicationSet deployer"},
+				},
+			},
+			valid: false,
+		},
+		{
+			name: "two deployers with different names but same configuration",
+			config: &Config{
+				Deployers: []Deployer{
+					{
+						Name:        "disapp",
+						Type:        "disapp",
+						Description: "Discovered Application deployer for discovered applications",
+					},
+					{
+						Name:        "disapp-recipe",
+						Type:        "disapp",
+						Description: "Discovered Application deployer for discovered applications with recipe",
+					},
+				},
+			},
+			valid: false,
+		},
+		{
+			name: "invalid type",
+			config: &Config{
+				Deployers: []Deployer{
+					{Name: "appset", Type: "appset", Description: "ApplicationSet deployer for ArgoCD"},
+					{Name: "subscr", Type: "subscr", Description: "Subscription deployer for OCM subscriptions"},
+					{Name: "disapp", Type: "discoapp", Description: "Discovered Application deployer for discovered applications"},
```

Comment:
For invalid name it will be more clear to use:

    {Name: "invalid", Type: "invalid", Description: "Invalid deployer"}

But we can improve this later.